tests/unit/test_meta_support.py

```
@@ -0,0 +1,150 @@
+        {
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "tools/call",
+            "_meta": {"progressToken": "pt-1"},
+            "params": {"name": "demo", "_meta": {"client": "test"}},
+        }
+    )
+
+
+def test_validate_request_meta_rejects_invalid_params_meta():
+    with pytest.raises(MetaValidationError, match="params._meta"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)